node/utils/src/lib.rs

```diff
@@ -66,3 +66,29 @@ where
     let new_struct: S = toml::from_str(toml_string)?;
     Ok(new_struct)
 }
+
+/// Get file count in a certain directory
+/// Will return the number of files in the directory
+///
+/// # Panics
+/// It will panic if:
+/// - The provided path doesn't exist.
+/// - The process lacks permissions to view the contents.
+/// - The path points at a non-directory file.
+///
+/// # Example
+/// ```
+/// use utils::count_files;
+/// use utils::get_home_dir;
+///
+/// let path_to_keystore = get_home_dir() + "/.forest/libp2p/keypair";
+/// let file_count = count_files(path_to_keystore[0..path_to_keystore.len()-8].to_string());
+/// assert_eq!(file_count, 1);
+/// ```
+pub fn count_files(dir: String) -> Result<u32> {
+    let mut file_count = 0;
+    for _i in read_dir(dir).expect("read files in the directory") {
+        file_count += 1;
+    }
+    Ok(file_count)
+}
```

Comment:
The return type is a result, but you're not passing up the error and instead panicing. The dir parameter does not need to be a String (heap allocated) and can instead be a `&str`. No need for for loop when only taking count.

Consider possibly something like: 
```
pub fn count_files(dir: &str) -> Result<usize> {
    Ok(read_dir(dir)?.enumerate().count())
}
```

Reply:
ok